src/registrar/models/domain_request.py

```diff
@@ -655,23 +661,62 @@ def save(self, *args, **kwargs):
-        reason_changed = self._cached_action_needed_reason != self.action_needed_reason
-        if was_already_action_needed and reason_exists and reason_changed:
-            # We don't send emails out in state "other"
-            if self.action_needed_reason != self.ActionNeededReasons.OTHER:
-                self._send_action_needed_reason_email(email_content=self.action_needed_reason_email)
+        self._cache_status_and_status_reasons()
+
+    def send_custom_status_update_email(self, status):
+        """Helper function to send out a second status email when the status remains the same,
+        but the reason has changed."""
+
+        # Currently, we store all this information in three variables.
+        # When adding new reasons, this can be a lot to manage so we store it here
+        # in a centralized location. However, this may need to change if this scales.
+        status_information = {
+            self.DomainRequestStatus.ACTION_NEEDED: {
+                "cached_reason": self._cached_action_needed_reason,
+                "reason": self.action_needed_reason,
+                "email": self.action_needed_reason_email,
+                "excluded_reasons": [DomainRequest.ActionNeededReasons.OTHER],
+                "wrap_email": True,
+            },
+            self.DomainRequestStatus.REJECTED: {
+                "cached_reason": self._cached_rejection_reason,
+                "reason": self.rejection_reason,
+                "email": self.rejection_reason_email,
+                "excluded_reasons": [],
+                # "excluded_reasons": [DomainRequest.RejectionReasons.OTHER],
+                "wrap_email": False,
+            },
+        }
+        status_info = status_information.get(status)
+
+        # Don't send an email if there is nothing to send.
+        if status_info.get("email") is None:
+            logger.warning("send_custom_status_update_email() => Tried sending an empty email.")
+            return
+
+        # We should never send an email if no reason was specified.
+        # Additionally, Don't send out emails for reasons that shouldn't send them.
+        if status_info.get("reason") is None or status_info.get("reason") in status_info.get("excluded_reasons"):
+            return
+
+        # Only send out an email if the underlying reason itself changed or if no email was sent previously.
+        if status_info.get("cached_reason") != status_info.get("reason") or status_info.get("cached_reason") is None:
+            bcc_address = settings.DEFAULT_FROM_EMAIL if settings.IS_PRODUCTION else ""
+            self._send_status_update_email(
+                new_status=status,
+                email_template="emails/includes/custom_email.txt",
+                email_template_subject="emails/status_change_subject.txt",
+                bcc_address=bcc_address,
+                custom_email_content=status_info.get("email"),
+                wrap_email=status_information.get("wrap_email"),
+            )
```

Comment:
OPTIONAL: Do we want an else block here that logs that send_custom_status_update_email met unexpected conditions?

Reply:
The function gets called. Inside the function, stuff happens only when certain conditions are right, otherwise it's just a ghost function silently running with no effect. I think we should log something to the effect of 'ran silently' which indicates an issue with our app or data